model-optimizer/extensions/middle/SplitConcatPairToInterpolate.py

```diff
@@ -53,7 +66,16 @@ def get_concat_after_split(split: Node) -> Optional[Node]:
 
     dest = split.out_port(0).get_destinations()[0].node
     # The transformation is applicable, only if next node is Concat.
-    return dest if dest.soft_get('type') == 'Concat' else None
+    if dest.soft_get('type') != 'Concat':
+        return
+
+    # The transformation is applicable, only if Split is a unique producer for Concat.
+    dest_inputs = [p.get_connection().get_source().node for _, p in dest.in_ports().items() if not p.disconnected()]
```

Comment:
This line can be simplified to:
`dest_inputs = [p.get_source().node for p in dest.in_ports().values() if not p.disconnected()]`

Reply:
Fixed.